src/Promise.js

```diff
@@ -154,23 +154,29 @@ if (!window.Promise) {
             }
         },
         fulfill: function (value) {
+            function deferredCallback(sub) {
+                return function() {
+                    sub.resolveCallback(value);
+                    promise.settledValue = value;
+                };
+            }
             var promise = this, subscriber;
             while (this.subscriberqueue.length) {
                 subscriber = this.subscriberqueue.shift();
-                setTimeout(function () {
-                    subscriber.resolveCallback(value);
-                    promise.settledValue = value;
```

Comment:
Does it matter that the settled value is reassigned per subscriber do you reckon?